### PR TITLE
backport VRF support from equinix/equinix provider

### PR DESCRIPTION
**Pull request by displague**
This is a backport of equinix/terraform-provider-equinix#129

Mostly derived from
https://patch-diff.githubusercontent.com/raw/equinix/terraform-provider-equinix/pull/129.patch
with some automated changes to the patch and manually cherry-picked changes from the rejects.

```
# vim commands applied against the patch
:%s_equinix/_metal/
:%s/equinix_//
:%s/data_source_metal/datasource_metal
:%s/_acc_test.go/_test.go
:%s/package equinix/package metal
:%s#docs/resources/metal_#docs/resources/
:%s#docs/data-sources/metal_#docs/data-sources/
```

Patch applied with:

```
patch -p1 --input=129.patch --fuzz=16
```

Rejects reviewed with:
```
find . -name \*rej  -exec less {} \;
```

Notably, rejects that dealt with test locations and plans were ignored for this PR.
Some helper tools for set/list conversion and sweeper test detection where copied from equinix/equinix.

### CODE REVIEW

File: metal/resource_metal_reserved_ip_block_test.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ resource "metal_project" "foobar" {
 
 resource "metal_reserved_ip_block" "test" {
 	project_id  = metal_project.foobar.id
-	facility    = "ny5"
+	facility    = "sv"
```
Comment (displague) [inline metal/resource_metal_reserved_ip_block_test.go:38]: This should be `metro =` (and line 104 should be metro too). Similar changes appear elsewhere.

File: metal/resource_metal_reserved_ip_block_test.go [hunk 2]
```diff
@@ -230,15 +230,15 @@ resource "metal_project" "foobar" {
 
 resource "metal_reserved_ip_block" "test" {
 	project_id  = metal_project.foobar.id
-	facility    = "ny5"
+	facility    = "ewr1"
```
Comment (displague) [inline metal/resource_metal_reserved_ip_block_test.go:233]: We would want to go from ewr1 to ny5, not reverse.

File: metal/resource_metal_spot_market_request_test.go [hunk 3]
```diff
@@ -17,8 +17,8 @@ resource "metal_project" "test" {
 }
 
 data "metal_spot_market_price" "test" {
-  facility = "sv15"
-  plan     = "c3.small.x86"
+  facility = "ewr1"
```
Comment (displague) [inline metal/resource_metal_spot_market_request_test.go:20]: sv15 would be preferable here. I think c3.small would be preferable as well.

File: metal/resource_metal_reserved_ip_block_test.go (hunk 2 above)
Comment (ocobleseqx) [reply metal/resource_metal_reserved_ip_block_test.go]: metros and facilities have been replaces with da/da11

File: metal/resource_metal_spot_market_request_test.go (hunk 3 above)
Comment (ocobleseqx) [reply metal/resource_metal_spot_market_request_test.go]: replaced with c3.medium

**Pull request merged**